### DISCUSSION TITLE
Feature Request - Lights - Favorite color profiles

### DISCUSSION THREAD

**Discussion by talormanda** [home-assistant/frontend#20125] [Dashboard]
I often run into an issue where I have many lights, and many light groups. When I want to quickly set the color, it defaults to this boring selection:
![image](https://github.com/home-assistant/frontend/assets/10092964/8efc5e30-ed2b-422e-ad18-5651c3c33b3b)

What I really want, is a new section where you can define your own color profiles, and select it from a drop-down menu, so my favorite colors across any light / group will look like this:
![image](https://github.com/home-assistant/frontend/assets/10092964/8b0202c9-9da5-40ce-aea7-f5b1a534aa3a)

Does anyone else run into this problem? I have too many lights to re-create this hundreds of times.

**Comment by Mariusthvdb** [comment]
given some discussion in Discord Frontend, I can see the use for this, and indeed +1 to setting our own defaults, and maybe even have those theme able.

Ive added some and notice this being added on the core.entty_registry on that light entity:
```
"light":{"favorite_colors":[{"color_temp_kelvin":2000},{"color_temp_kelvin":3512},{"color_temp_kelvin":5023},{"color_temp_kelvin":6535},{"rgb_color":[127,172,255]},{"rgb_color":[215,150,255]},{"rgb_color":[255,158,243]},{"rgb_color":[255,110,84]},{"hs_color":[358,50]},{"color_temp_kelvin":2135}]}}
```

(doesnt go away after deleting the manually added ones.....) so once you add, the line is there to stay

could be copied rather easily though (not advising to do so, but just sayin)

adding color names would be nice too

**Reply by talormanda** [reply]
I agree. I would like to see the ability to build out color profiles you can pick from, and even taking one of them and "setting it as the default". So you can still pick any profile you want to view and change a color with inside there, but the default is also something you "set".

**Comment by kristof-dragon** [comment]
+1 for this request from me as well, as I was looking to template the default colours/selections, but was not able to find anything yet.

**Comment by ebozonne** [comment]
+1
And strange that there is a list for the effects and not the colors, why?
Anyway, I found a work around for my automations to make a permutation between colors, it was not easy at all because lights take a random approximation of the color you send... (I had to make a test with a tolerance, but if anyone has a better option it would be cool)
-> [https://community.home-assistant.io/t/is-it-possible-to-modify-or-trigger-lights-favorite-colors-via-yaml-or-the-ui/820844/2?u=ebozonne](https://community.home-assistant.io/t/is-it-possible-to-modify-or-trigger-lights-favorite-colors-via-yaml-or-the-ui/820844/2?u=ebozonne)

**Comment by j9brown** [comment]
As far as I can tell, there is no service for setting favorite colors.  They're only ever accessed from the front end.

However, favorite colors appear to be stored in the light's entity options in the entity registry and so it should be possible to make a service to manipulate them with [async_update_entity_options](https://github.com/home-assistant/core/blob/35416189f2944f58b5b91d9cf8ac90a491a10890/homeassistant/helpers/entity_registry.py#L1315).  I'm planning to give this a try shortly.

**Comment by j9brown** [comment]
I wrote an integration that lets you set and get favorite colors and configure them in YAML: https://community.home-assistant.io/t/scenery-better-light-profiles/883381

Enjoy!
